llvm/docs/LangRef.rst

```diff
@@ -24975,6 +24975,72 @@ The '``llvm.masked.compressstore``' intrinsic is designed for compressing data i
 
 Other targets may support this intrinsic differently, for example, by lowering it into a sequence of branches that guard scalar store operations.
 
+Masked Vector Compress Intrinsic
```

Comment:
This might be bikeshedding, but would it make more sense to put this in the llvm.vector.* family of intrinsics? Since otherwise this would be the only masked intrinsic that doesn't read or write memory

Reply:
Sounds good. I've moved it to the `vector.*` section.

Reply:
After re-reading this, I think I misunderstood your comments, sorry. You were advocating for calling the the intrinsic `@llvm.vector.compress` instead of `@llvm.masked.compress` and not just moving this in the docs? @lukel97 @nikolaypanchenko Could you please confirm this?

Reply:
Yeah that's what I originally had in mind, but I'm not strongly opinionated so I'll leave this up to you/other reviewers. This PR looks like it's making good progress and I don't want to block it :)

Reply:
The more I think about it, the more I think you are right about the naming. The "mask" is an essential part of the operation, as @nikolaypanchenko points out, so it does not make sense to use it to distinguish between a masked and non-masked compress. 

Pulling in @efriedma-quic @RKSimon for an opinion on naming and putting it in the experimental namespace.

If I change it, I'd probably change it to `@llvm.experimental.vector.compress` as suggested by @lukel97 , as this covers both the `vector` and `experimental` (see [other comment](https://github.com/llvm/llvm-project/pull/92289/#discussion_r1646842999) for discussion on this) for now.